libp2p/protocols/identify.nim

```diff
@@ -150,24 +157,26 @@ proc identify*(p: Identify,
     raise newException(IdentityInvalidMsgError, "Incorrect message received!")
   result = infoOpt.get()
 
-  if not isNil(remotePeerInfo) and result.pubKey.isSome:
+  if result.pubKey.isSome:
     let peer = PeerID.init(result.pubKey.get())
     if peer.isErr:
       raise newException(IdentityInvalidMsgError, $peer.error)
     else:
-      if peer.get() != remotePeerInfo.peerId:
+      result.peerId = peer.get()
+      if peer.get() != remotePeerId:
         trace "Peer ids don't match",
               remote = peer,
-              local = remotePeerInfo.peerId
+              local = remotePeerId
 
         raise newException(IdentityNoMatchError, "Peer ids don't match")
+  else:
+    raise newException(IdentityInvalidMsgError, "No pubkey in identify")
 
-proc new*(T: typedesc[IdentifyPush], connManager: ConnManager): T =
-  let identifypush = T(connManager: connManager)
+proc new*(T: typedesc[IdentifyPush], handler: IdentifyPushHandler = nil): T =
```

Comment:
Should we split this out into it's own PR?

Reply:
It would duplicate a bit of work, since this PR completely breaks the identify process anyway
And right now nobody uses the IdentifyPush, so should be ok to modify it "sneakily"

But I can split if you want